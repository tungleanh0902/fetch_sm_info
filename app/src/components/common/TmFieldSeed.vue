<template>
  <TmField
    :value="value"
    class="tm-field-seed"
    type="textarea"
    @input="update($event)"
  />
</template>

<script>
import autosize from "autosize"
import TmField from "common/TmField"
export default {
  name: `tm-field-seed`,
  components: { TmField },
  props: {
    value: {
      type: String,
      required: true,
    },
  },
  watch: {
    value: {
      handler: async function () {
        await this.$nextTick()
        autosize.update(this.$el)
      },
    },
  },
  mounted() {
    // adjust the textarea element height to match content
    autosize(this.$el)
  },
  methods: {
    update(value) {
      this.$emit(`input`, value)
    },
  },
}
</script>
<style scoped>
.tm-field:disabled.tm-field-seed {
  border: 1px solid var(--bc);
  line-height: 22px;
  padding: 0.5rem;
  min-height: 90px;
}

.field-seed {
  line-height: var(--lg);
  padding: 0.75rem;
  resize: none;
}
</style>
