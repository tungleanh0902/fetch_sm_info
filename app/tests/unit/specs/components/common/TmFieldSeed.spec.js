import { mount } from "@vue/test-utils"
import TmFieldSeed from "common/TmFieldSeed"

describe(`TmFieldSeed`, () => {
  let wrapper

  const propsData = {
    value: `one two three four five six seven eight nine ten eleven twelve`,
  }

  beforeEach(() => {
    wrapper = mount(TmFieldSeed, { propsData })
  })

  it(`has a value from props`, () => {
    expect(wrapper.vm.value).toContain(
      `one two three four five six seven eight nine ten eleven twelve`
    )
  })

  it(`has the expected html structure`, () => {
    expect(wrapper.element).toMatchSnapshot()
  })

  it(`has the correct class`, () => {
    expect(wrapper.find(`.tm-field`).classes()).toContain(`tm-field-seed`)
  })

  it(`emits input event on update method call`, () => {
    wrapper.vm.update(`test event`)
    expect(wrapper.emitted().input[0][0]).toEqual(`test event`)
  })

  it(`does not emit input event when update method not called`, () => {
    expect(wrapper.emitted()).toEqual({})
  })
  it(`does not call autosize.update when value does not change`, async () => {
    const autosize = require(`autosize`)
    const spy = jest.spyOn(autosize, `update`)
    expect(spy.mock.calls).toEqual([])
  })
  it(`calls autosize.update when value changes`, async () => {
    const autosize = require(`autosize`)
    const spy = jest.spyOn(autosize, `update`)
    wrapper.setProps({ value: `change value` })
    await wrapper.vm.$nextTick()
    expect(spy).toBeCalled()
    expect(spy.mock.calls[0].length).toEqual(1)
  })
})
