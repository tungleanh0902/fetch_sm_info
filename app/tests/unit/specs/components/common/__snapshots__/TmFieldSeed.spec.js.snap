// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TmFieldSeed has the expected html structure 1`] = `
<textarea
  class="tm-field-seed tm-field"
  style="overflow-x: hidden; word-wrap: break-word; overflow-y: hidden;"
/>
`;
